Source/WebCore/platform/ios/UserAgentIOS.mm

```diff
@@ -95,9 +97,12 @@ String standardUserAgentWithApplicationName(const String& applicationName, const
     UNUSED_PARAM(separator);
     return makeString("Mozilla/5.0 (iPad; CPU OS 16_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.3 Mobile/15E148 Safari/604.1");
 #else
-    // FIXME: We should deprecate and remove this override; see https://bugs.webkit.org/show_bug.cgi?id=217927 for details.
-    if (auto override = dynamic_cf_cast<CFStringRef>(adoptCF(CFPreferencesCopyAppValue(CFSTR("UserAgent"), CFSTR("com.apple.WebFoundation")))))
-        return override.get();
+    if (!linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::DoesNotOverrideUAFromNSUserDefult)) {
+        if (auto override = dynamic_cf_cast<CFStringRef>(adoptCF(CFPreferencesCopyAppValue(CFSTR("UserAgent"), CFSTR("com.apple.WebFoundation"))))) {
+            NSLog(@"Reading an override UA from the NSUserDefault [com.apple.WebFoundation UserAgent]. This is incompatible with the modern need to compose the UA and clients should use the API to set the application name or UA instead.");
```

Comment:
Usually we set a static and only log once in cases like this, don't want to cause log spam.

Reply:
Fixed with the `static BOOL hasLoggedDeprecationWarning` in https://github.com/WebKit/WebKit/commit/1d8d75f9f783f278033bb894174663bb1fe99a6d